packages/property-grid/src/components/functional/FunctionalElementList.tsx

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import "../ElementList.scss";
+
+import { PresentationLabelsProvider } from "@bentley/presentation-components";
+import {
+  RowItem,
+  SelectionMode,
+  SimpleTableDataProvider,
+  Table,
+} from "@bentley/ui-components";
+import { Icon } from "@bentley/ui-core";
+import classnames from "classnames";
+import * as React from "react";
+
+import { PropertyGridManager } from "../../PropertyGridManager";
+import { createDataProvider, ElementListProps } from "../ElementList";
+
+export const FunctionalElementList = ({
+  iModelConnection,
+  instanceKeys,
+  onBack,
+  onSelect,
+  rootClassName,
+}: ElementListProps) => {
+  const [dataProvider, setDataProvider] =
+    React.useState<SimpleTableDataProvider>();
+
+  const labelsProvider: PresentationLabelsProvider = React.useMemo(() => {
+    return new PresentationLabelsProvider({
+      imodel: iModelConnection,
+    });
+  }, [iModelConnection]);
+
+  React.useEffect(() => {
+    const onMount = async () => {
```

Comment:
this could/will execute more often than on mount....not sure the name makes sense?